docs/metrics.md

```diff
@@ -1,37 +1,140 @@
-- `nodejs.memory.gc.size` (cumulative_counter, bytes) - Total collected by the garbage collector.
-- `nodejs.memory.gc.pause` (cumulative_counter, nanoseconds) - Time spent doing GC.
-- `nodejs.memory.gc.count` (cumulative_counter, count) - Number of times GC ran.
-- `nodejs.event_loop.lag.max` (gauge, nanoseconds) - Max event loop lag within the collection interval.
-- `nodejs.event_loop.lag.min` (gauge, nanoseconds) - Min event loop lag within the collection interval.
+Custom exporters and/or readers can be provided via the `metricReaderFactory` option.
 
-## Custom metrics
+:warning: Setting `metricReaderFactory` will invalidate `exportInterval` and `endpoint` options.
 
-You can send custom metrics via the internal [SignalFx client](https://github.com/signalfx/signalfx-nodejs#reporting-data):
+```javascript
+const { startMetrics } = require('@splunk/otel');
+const { PrometheusExporter } = require('@opentelemetry/exporter-prometheus');
+const { OTLPMetricExporter } = require('@opentelemetry/exporter-metrics-otlp-http');
+const { PeriodicExportingMetricReader } = require('@opentelemetry/sdk-metrics-base');
+
+startMetrics({
+  serviceName: 'my-service',
+  metricReaderFactory: () => {
+    return [
+      new PrometheusExporter(),
+      new PeriodicExportingMetricReader({
+        exportIntervalMillis: 1000,
+        exporter: new OTLPMetricExporter({ url: 'http://localhost:4318' })
+      })
+    ]
+  }
+});
+```
+
+## Choosing temporality aggregation type
 
 ```javascript
 const { startMetrics } = require('@splunk/otel');
+const { OTLPMetricExporter } = require('@opentelemetry/exporter-metrics-otlp-grpc');
+const { AggregationTemporality, PeriodicExportingMetricReader } = require('@opentelemetry/sdk-metrics-base');
+
+startMetrics({
+  serviceName: 'my-service',
+  metricReaderFactory: () => {
+    return [
+      new PeriodicExportingMetricReader({
+        exporter: new OTLPMetricExporter({
+          temporalityPreference: AggregationTemporality.DELTA
+        })
+      })
+    ]
+  }
+});
+```
+
+## Migrating from SignalFx metrics
+
+The SignalFx client is no longer available.
 
-const { getSignalFxClient } = startMetrics();
+```javascript
+// Before
+const { startMetrics } = require('@splunk/otel');
+const { getSignalFxClient } = startMetrics({ serviceName: 'my-service' });
 
 getSignalFxClient().send({
-  gauges: [{ metric: 'my.app.foo', value: 42, timestamp: 1442960607000}]
+  gauges: [{ metric: 'cpu', value: 42, timestamp: 1442960607000}],
+  cumulative_counters: [{ metric: 'clicks', value: 99, timestamp: 1442960607000}],
 })
+
+// After
+const { startMetrics } = require('@splunk/otel');
+const { metrics } = require('@opentelemetry/api-metrics');
+
+startMetrics({ serviceName: 'my-service' });
+
+const meter = metrics.getMeter('my-meter');
+meter.createObservableGauge('cpu', result => {
+  result.observe(42);
+});
+const counter = meter.createCounter('clicks');
+counter.add(99);
+```
+
+## Runtime metrics
```

Comment:
@seemk @Rauno56 I understand memory profiling metrics are still to be named?

Reply:
Current list of metrics that will be used by memory profiling dashboard:
- `process.runtime.nodejs.memory.heap.total`
- `process.runtime.nodejs.memory.heap.used`
- `process.runtime.nodejs.memory.rss`
- `process.runtime.nodejs.memory.gc.pause`
- `process.runtime.nodejs.memory.gc.count`

Plus there might be another metric in the future that currently has no implementation: allocations / allocation rate. Will be added in a separate PR if it's possible to get the data.